java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -248,8 +263,21 @@ private static boolean isVolatileField(Symbol symbol) {
     }
   }
 
-  private static boolean isDisposable(SymbolicValue symbolicValue, @Nullable Object constraint) {
-    return SymbolicValue.isDisposable(symbolicValue) && (constraint == null || !(constraint instanceof ObjectConstraint) || ((ObjectConstraint) constraint).isDisposable());
+  private static boolean isDisposable(SymbolicValue symbolicValue, @Nullable Constraint constraint) {
+    //FIXME this should be handle with callbacks rather than keeping those value in programstate
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)